### PR TITLE
Add support for ACS

### LINKED ISSUE #324: Add support for Alternate Content Sources

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #324

### CODE REVIEW

File: pulpcore/cli/file/acs.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+
+remote_option = resource_option(
+    "--remote",
+    default_plugin="file",
+    default_type="file",
+    context_table={"file:file": PulpFileRemoteContext},
+    href_pattern=PulpRemoteContext.HREF_PATTERN,
+    help=_(
+        "Remote used for synching in the form '[[<plugin>:]<resource_type>:]<name>' or by href."
```
Comment (mdellweg) [inline pulpcore/cli/file/acs.py:91]: "Remote this ACS is attached to..." ?

File: tests/scripts/pulp_file/test_acs.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+cleanup
+
+expect_succ pulp file remote create --name $acs_remote --url "http://example.com" --policy "on_demand"
+
+expect_succ pulp file acs create --name "cli_test_acs" --remote $acs_remote
+expect_succ pulp file acs list
+test "$(echo "$OUTPUT" | jq -r length)" -gt 1
+expect_succ pulp file acs show --name "cli_test_acs"
+expect_succ pulp file acs destroy --name "cli_test_acs"
```
Comment (mdellweg) [inline tests/scripts/pulp_file/test_acs.sh:24]: Can we add some commands with "--path" here?

File: pulpcore/cli/file/acs.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+@acs.group()
+def path() -> None:
+    """Manage paths for an ACS."""
+    pass
+
+
+@path.command()
+@name_option
+@href_option
+@click.option("--path", "paths", multiple=True, help=_("path to add to ACS"))
```
[the remark this answers is not in the document]
Comment (daviddavis) [reply pulpcore/cli/file/acs.py:55]: I wonder if we can do this automatically. Maybe something to think about..

**Comment by daviddavis** [comment]
I think this PR can be merged. I'm working the refresh commands and handling the TaskGroup responses that the refresh endpoints will return but I'll open follow up PRs for that.

**Comment by mdellweg** [comment]
> I think this PR can be merged. I'm working the refresh commands and handling the TaskGroup responses that the refresh endpoints will return but I'll open follow up PRs for that.

Until then, the commands would return the reference to the task-group, right?

**Review by mdellweg — APPROVED** [review]

**Comment by daviddavis** [comment]
That would be my assumption but since there's no endpoints currently that return a task group, I haven't tested it out yet.

**Pull request merged**